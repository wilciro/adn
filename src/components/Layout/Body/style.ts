import styled from 'styled-components';

interface BodyProps {
  dark: boolean;
}

export const Body = styled.div<BodyProps>`
  width: 80%;
  padding: 80px 15px;
  border-radius: 10px;
  margin: 20px auto;
  min-height: 100vh;

  @media (max-width: 768px) {
    width: 100%;
  }
`;
