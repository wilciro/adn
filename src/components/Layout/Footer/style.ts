import styled from "styled-components";

export const Footer = styled.header`
    width: 90%;
    padding: 10px 15px;
    margin: 20px auto;
    text-align: center;
`;
