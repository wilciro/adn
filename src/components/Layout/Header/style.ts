import styled from 'styled-components';

export const Header = styled.header`
  width: 90%;
  padding: 10px 15px;
  border-radius: 10px;
  box-shadow: 0 0 2px rgba(10, 10, 10, 0.5);
  background: rgba(240, 240, 240, 0.7);
  backdrop-filter: blur(5px);
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;
  position: fixed;
  top: 5px;
  left: 5%;
  z-index: 5;

  @media (max-width: 768px) {
    width: 100%;
    left: 0px;
    top: 0px;
  }
`;

export const ButtonsSeparator = styled.div`
  display: flex;
  align-items: center;
  gap: 10px;
`;

interface NavProps {
  open: boolean;
}

export const Nav = styled.nav<NavProps>`
  width: 100%;
  height: 100vh;
  position: fixed;
  top: 0px;
  background: rgba(10, 10, 10, 0.8);
  backdrop-filter: blur(10px);
  z-index: 10;
  display: ${(p) => (p.open ? 'flex' : 'none')};
  align-items: center;
`;

export const NavMenu = styled.ul`
  list-style: none;
  font-size: 8vw;
  margin-left: 5%;

  a {
    color: white;
    text-decoration: none;
    transition: all 0.3s;
    &:hover {
      opacity: 0.9;
      color: var(--primary-color);
    }
  }
`;

export const NavClose = styled.button`
  position: fixed;
  top: 10px;
  right: 10px;
  font-size: 4em;
  width: 100px;
  height: 100px;
  border-radius: 100%;
  cursor: pointer;
  background: none;
  color: white;
  border: none;
  &:hover {
    opacity: 0.8;
    color: var(--primary-color);
  }
`;
