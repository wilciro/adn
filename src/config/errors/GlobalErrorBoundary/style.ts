import styled from "styled-components";

export const Container = styled.div`
    width: 60%;
    padding: 10px 15px;
    margin: 20px auto;
    text-align: center;
    height: 100vh;
    display: flex;
    align-items: center;
`;