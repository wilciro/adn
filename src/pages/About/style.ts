import styled from 'styled-components';

export const AboutDiv = styled.section`
  width: 96%;
  margin: auto;
  display: flex;
  justify-content: space-between;
  align-items: center;

  img {
    width: 40%;
  }

  @media (max-width: 768px) {
    width: 100%;
    display: block;
    img {
      width: 70%;
      margin: auto;
      display: block;
    }
  }
`;
