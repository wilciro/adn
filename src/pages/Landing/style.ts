import styled from 'styled-components';

export const CheckoutDiv = styled.section`
  width: 300px;
  margin: auto;
  border: solid 1px rgb(230, 230, 230);
  border-radius: 10px;
  padding: 5px 25px;
  position: fixed;
  right: 10px;
  top: 100px;
  z-index: 10;
  background: white;

  @media (max-width: 1000px) {
    width: 100%;
    display: block;
    bottom: 0px;
    top: auto;
    right: auto;
    left: 0px;
    z-index: 3;
    background: rgba(255, 255, 255, 0.6);
    backdrop-filter: blur(5px);
  }
`;
