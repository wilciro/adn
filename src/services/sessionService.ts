import { v4 as uuidv4 } from 'uuid';
import { SessionState } from 'context/models/SessionState';

const SESSION_IDENTIFIER = 'SESSION_ID';
const SESSION_SEPARATOR = ':::';

export const createSession = (username?: string): string => {
  const sessionId = `${uuidv4()}${SESSION_SEPARATOR}${username}${SESSION_SEPARATOR}${Date.now()}`;
  localStorage.setItem(SESSION_IDENTIFIER, sessionId);
  return sessionId;
};
export const recoverSession = (): SessionState => {
  const sessionId = localStorage.getItem(SESSION_IDENTIFIER);
  if (!sessionId) {
    return {};
  }
  return {
    sessionId,
    username: sessionId.split(SESSION_SEPARATOR)[1],
  };
};
