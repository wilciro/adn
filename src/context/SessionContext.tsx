import React, { createContext, useEffect, useRef, useState } from 'react';
import { SessionState } from 'context/models/SessionState';
import * as sessionService from '../services/sessionService';

export const useStateContainer = (initialState: SessionState = {}) => {
  const [username, setUsername] = useState(initialState.username);
  const [sessionId, setSessionId] = useState(initialState.sessionId);

  useEffect(() => {
    if (!username) {
      setSessionId(sessionService.deleteSession());
      return;
    }
    setSessionId(sessionService.createSession(username));
  }, [username]);

  return {
    data: { username, sessionId },
    mutations: { setUsername },
  };
};

export const SessionContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

interface SessionProps {
  children: React.ReactNode;
}

export const SessionProvider = ({ children }: SessionProps) => {
  const contextValue = useStateContainer(sessionService.recoverSession());
  return (
    <SessionContext.Provider value={contextValue}>
      {children}
    </SessionContext.Provider>
  );
};

