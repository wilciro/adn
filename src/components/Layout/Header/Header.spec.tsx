import React from 'react';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import { SessionProvider } from 'context/SessionContext';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import * as sessionService from '../../../services/sessionService';
import { Header } from './style';
import LayoutHeader from '.';

describe('Header tests', () => {
  test('should match style', () => {
    const { container } = render(<Header>Ceiba</Header>);
    expect(container).toMatchSnapshot();
  });
  test('should match snapshot Header', () => {
    const { container } = render(
      <BrowserRouter>
        <SessionProvider>
          <LayoutHeader />
        </SessionProvider>
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });
  test('should open menu', () => {
    const { container } = render(
      <BrowserRouter>
        <SessionProvider>
          <LayoutHeader />
        </SessionProvider>
      </BrowserRouter>,
    );
    // btn-nav-menu
    fireEvent.click(screen.getByTestId('btn-nav-menu'));
    expect(screen.getByTestId('nav-menu')).toHaveStyle(`display: flex`);
  });
  test('should logout', async () => {
    sessionService.createSession('admin');
    const { container } = render(
      <BrowserRouter>
        <SessionProvider>
          <LayoutHeader />
          <Routes>
            <Route path="/login" element={<h2 data-testid="login">Login</h2>} />
            <Route path="/" element={<h2 data-testid="landing">Landing</h2>} />
          </Routes>
        </SessionProvider>
      </BrowserRouter>,
    );
    // btn-nav-menu
    fireEvent.click(screen.getByTestId('header-logout-link'));
    await waitFor(() => {
      expect(screen.getByTestId('login')).toBeInTheDocument();
    });
  });
});
