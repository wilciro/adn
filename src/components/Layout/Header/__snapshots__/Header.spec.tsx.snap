// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should match snapshot Header 1`] = `
<div>
  <header
    class="sc-bczRLJ eJfard"
  >
    <div
      class="sc-gsnTZi hpkuKe"
    >
      <button
        class="mantine-ActionIcon-hover mantine-ActionIcon-root mantine-t0y1bq"
        data-testid="btn-nav-menu"
        title="MenÃº"
        type="button"
      >
        <svg
          class="icon icon-tabler icon-tabler-menu-2"
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <line
            x1="4"
            x2="20"
            y1="6"
            y2="6"
          />
          <line
            x1="4"
            x2="20"
            y1="12"
            y2="12"
          />
          <line
            x1="4"
            x2="20"
            y1="18"
            y2="18"
          />
        </svg>
      </button>
      <img
        alt="Logo"
        src="/assets/192.png"
      />
    </div>
    <div
      class="sc-gsnTZi hpkuKe"
    >
      <a
        href="/login"
        id="header-login-link"
      >
        <button
          class="mantine-Button-filled mantine-Button-root mantine-1grzg0q"
          type="button"
        >
          <div
            class="mantine-3xbgk5 mantine-Button-inner"
          >
            <span
              class="mantine-qo1k2 mantine-Button-label"
            >
              Ingresar
            </span>
          </div>
        </button>
      </a>
    </div>
  </header>
  <nav
    class="sc-dkzDqf JmMFC"
    data-testid="nav-menu"
  >
    <button
      class="sc-eCYdqJ kgffIQ"
    >
      X
    </button>
    <ul
      class="sc-hKMtZM jUMTUe"
    >
      <li>
        <a
          data-testid="link-to-landing"
          href="/"
        >
          Inicio
        </a>
      </li>
      <li>
        <a
          data-testid="link-to-about"
          href="/about"
        >
          Acerca de la app
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Header tests should match style 1`] = `
<div>
  <header
    class="sc-bczRLJ eJfard"
  >
    Ceiba
  </header>
</div>
`;
